### ISSUE TITLE
SpaceAroundOperators fails on Foo[] call

### ISSUE THREAD

**Issue by mseppae** [bbatsov/rubocop#2774]
Faced this breakage within our code when initializing empty Hpricot elements array with Hpricot::Elements[]

Here is a failings sample

```ruby
class Foo
  def self.[](*args)
    p args
  end
end

Foo['a', 'b']
Foo[] # Fails
```

Console output
```
mseppae@Markos-MBP % rubocop test.rb
Inspecting 1 file
An error occurred while Style/SpaceAroundOperators cop was inspecting /Path/to/test.rb.
To see the complete backtrace run rubocop -d.
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/SpaceAroundOperators cop was inspecting /Path/to/test.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.37.0 (using Parser 2.3.0.2, running on ruby 2.1.7 x86_64-darwin14.0)
```
